File: app/javascript/channels/messages_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("MessagesChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    //alert('connect')
    roomId = $("#chat-box").data("room-id")
    checkIn(roomId)
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
    posts = $(".messages-row").length
    if(posts == 10){
      $(".message-row").first().remove()

      $("#chat-box").append(data)
      $("#message-field").val('')

    }
      
  },

  checkIn(roomId) {
    if(roomId){
      perform = 'checkIn'
      room_id = roomId
    }
    else{
      perform = 'checkOut'
    }
      
  }

});

File: app/channels/messages_channel.rb
class MessagesChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"

  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def checkIn(data)
    room_id = "#{data['room_id']}"
    stream_from "messages_room_#{room_id}"
  end

  def checkOut
    stop_all_streams
  end
end
